block-streamer/src/receiver.rs

```diff
@@ -1,73 +1,36 @@
 use anyhow::Result;
-use block_ship::{types::BlockWrapper, types::CID_MARKER_LEN};
+use cid::Cid;
 use local_storage::storage::{Storage, StoredBlock};
-use messages::{TransmissionChunk, TransmissionMessage};
-use std::collections::BTreeMap;
+use messages::{TransmissionBlock, TransmissionMessage};
 use std::rc::Rc;
-use tracing::info;
 
 pub struct Receiver {
-    // Map of cid_marker to cid
-    pub cids_to_build: BTreeMap<Vec<u8>, Vec<u8>>,
-    // Map of cid_marker to [chunk]
-    pub cid_chunks: BTreeMap<Vec<u8>, Vec<TransmissionChunk>>,
+    // Handle to Storage
     pub storage: Rc<Storage>,
 }
 
 impl Receiver {
     pub fn new(storage: Rc<Storage>) -> Receiver {
-        Receiver {
-            cids_to_build: BTreeMap::new(),
-            cid_chunks: BTreeMap::new(),
-            storage,
-        }
+        Receiver { storage }
     }
 
-    pub fn handle_chunk_msg(&mut self, chunk: TransmissionChunk) -> Result<()> {
-        info!(
-            "Received chunk {} for CID: {:?}",
-            chunk.chunk_offset, chunk.cid_marker
-        );
-
-        self.cid_chunks
-            .entry(chunk.cid_marker.clone())
-            .and_modify(|vec| vec.push(chunk.clone()))
-            .or_insert(vec![chunk.clone()]);
-        Ok(())
-    }
-
-    pub fn handle_cid_msg(&mut self, cid: Vec<u8>) -> Result<()> {
-        let cid_marker = &cid[..CID_MARKER_LEN];
-        self.cids_to_build.entry(cid_marker.to_vec()).or_insert(cid);
-        Ok(())
-    }
-
-    pub fn attempt_block_assembly(&mut self) -> Result<()> {
-        for (cid_marker, cid) in &self.cids_to_build {
-            if let Some(cid_chunks) = &self.cid_chunks.get(cid_marker) {
-                if let Ok(wrapper) = BlockWrapper::from_chunks(cid, cid_chunks) {
-                    let stored_block: StoredBlock = wrapper.try_into()?;
-                    info!(
-                        "Found block {} with {} bytes {} links",
-                        &stored_block.cid,
-                        &stored_block.data.len(),
-                        &stored_block.links.len()
-                    );
-                    self.storage.import_block(&stored_block)?;
-                    // Purge chunks after insert
-                    self.cid_chunks.remove(cid_marker);
-                }
-            }
+    pub fn handle_block_msg(&mut self, block: &TransmissionBlock) -> Result<()> {
+        let mut links = vec![];
+        for l in &block.links {
+            links.push(Cid::try_from(l.clone())?.to_string());
         }
-        Ok(())
+        let stored_block = StoredBlock {
+            cid: Cid::try_from(block.cid.clone())?.to_string(),
+            data: block.data.clone(),
```

Comment:
You can avoid this clone and just pass ownership of the block into this function.